Sources/IO/Geometry/GLTFImporter/Reader.js

```diff
@@ -383,37 +379,47 @@ function handlePrimitiveExtensions(extensions, model, node) {
  * @param {GLTFMesh} mesh - The GLTF mesh
  * @returns {vtkActor} The created VTK actor
  */
-async function createActorFromGTLFNode(model, node, worldMatrix) {
+async function createActorFromGTLFNode(worldMatrix) {
   const actor = vtkActor.newInstance();
   const mapper = vtkMapper.newInstance();
   mapper.setColorModeToDirectScalars();
   actor.setMapper(mapper);
   actor.setUserMatrix(worldMatrix);
 
-  if (node.mesh !== undefined) {
-    const polyData = await createPolyDataFromGLTFMesh(node.mesh);
-    mapper.setInputData(polyData);
+  const polydata = vtkPolyData.newInstance();
+  mapper.setInputData(polydata);
+  return actor;
+}
+
+/**
+ * Creates a VTK actor from a GLTF mesh
+ * @param {GLTFMesh} mesh - The GLTF mesh
+ * @returns {vtkActor} The created VTK actor
+ */
+async function createActorFromGTLFPrimitive(model, primitive, worldMatrix) {
+  const actor = vtkActor.newInstance();
+  const mapper = vtkMapper.newInstance();
+  mapper.setColorModeToDirectScalars();
+  actor.setMapper(mapper);
+  actor.setUserMatrix(worldMatrix);
 
-    const primitive = node.mesh.primitives[0]; // the first one for now
+  const polydata = await createPolyDataFromGLTFMesh(primitive);
+  mapper.setInputData(polydata);
 
-    // Support for materials
-    if (primitive.material !== undefined) {
-      await createPropertyFromGLTFMaterial(model, primitive.material, actor);
-    }
+  // Support for materials
+  if (primitive.material !== null) {
```

Comment:
Should be `primitive.material != null`.
Indeed, `primitive.material != null` covers `primitive.material !== null && primitive.material !== undefined`

Reply:
i have updated the code